mongo/client_examples_test.go

```diff
@@ -0,0 +1,200 @@
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = client
+}
+
+func ExampleConnect_direct() {
+	// Create a direct connection to a host. The driver will send all requests to that host and will not
+	// automatically discover other hosts in the deployment.
+
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017/?connect=direct")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = client
+}
+
+func ExampleConnect_sCRAM() {
+	// Configure a Client with SCRAM authentication (https://docs.mongodb.com/manual/core/security-scram/).
+	// The default authentication database for SCRAM is "admin". This can be configured via the
+	// database field in the URI or the AuthSource field in the options.Credential struct.
+	// SCRAM is the default auth mechanism so specifying a mechanism is not required.
+
+	// To configure auth via URI instead of a Credential, use
+	// "mongodb://user:password@localhost:27017".
+	credential := options.Credential{
+		Username: "user",
+		Password: "password",
+	}
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017").SetAuth(credential)
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = client
+}
+
+func ExampleConnect_x509() {
+	// Configure a Client with X509 authentication (https://docs.mongodb.com/manual/core/security-x.509/).
+
+	// X509 can be configured with different sets of options in the connection string:
+	// 1. tlsCAFile (or SslCertificateAuthorityFile): Path to the file with either a single or bundle of certificate
+	// authorities to be considered trusted when making a TLS connection.
+	// 2. tlsCertificateKeyFile (or SslClientCertificateKeyFile): Path to the client certificate file or the client
+	// private key file. In the case that both are needed, the files should be concatenated.
+
+	// The SetAuth client option should also be used. The username field is optional. If it is not specified, it will
+	// be extracted from the certificate key file. The AuthSource is required to be $external.
+
+	caFilePath := "path/to/cafile"
+	certificateKeyFilePath := "path/to/client-certificate"
+
+	// To configure auth via a URI instead of a Credential, append "&authMechanism=mongodb-x509" to the URI.
+	uri := "mongodb://host:port/?tlsCAFile=%s&tlsCertificateKeyFile=%s"
+	uri = fmt.Sprintf(uri, caFilePath, certificateKeyFilePath)
+	credential := options.Credential{
+		AuthMechanism: "MONGODB-X509",
+	}
+	clientOpts := options.Client().ApplyURI(uri).SetAuth(credential)
+
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = client
+}
+
+func ExampleConnect_pLAIN() {
+	// Configure a Client with LDAP authentication
+	// (https://docs.mongodb.com/manual/core/authentication-mechanisms-enterprise/#security-auth-ldap).
+	// MongoDB Enterprise supports proxy authentication through an LDAP service that can be used through the PLAIN
```

Comment:
Note here that this mechanism sends the password in plaintext and so should only be used in concert with TLS connections

Reply:
done